packages/server/src/core/internals/procedureBuilder.ts

```diff
@@ -252,6 +255,11 @@ export function createBuilder<TConfig extends AnyRootConfig>(
       return createNewBuilder(_def, builder._def) as any;
     },
     use(middleware) {
+      if ('pipe' in middleware) {
+        return createNewBuilder(_def, {
+          middlewares: middleware._self.piped ?? [middleware._self],
+        }) as AnyProcedureBuilder;
+      }
```

Comment:
I'd like to get rid of this conditional but I guess it's okay too 🤷

Reply:
yeah this part in particular hurts me.